pkg/k8sutil/node.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var hostRoot = os.Getenv("HOST_ROOT")
+
+// GetCurrentNode is the best effort to get the name of the current k8s node.
+func GetCurrentNode() (string, error) {
+	kubeconfig, err := detectKubeconfig()
+	if err != nil {
+		return "", fmt.Errorf("getting kubeconfig: %w", err)
+	}
+	clientset, err := NewClientset(kubeconfig)
+	if err != nil {
+		return "", fmt.Errorf("creating clientset: %w", err)
+	}
+	nodes, err := clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("listing nodes: %w", err)
+	}
+	machineId, err := getMachineID()
+	if err != nil {
+		return "", fmt.Errorf("getting machine-id: %w", err)
+	}
+	for _, node := range nodes.Items {
+		if node.Status.NodeInfo.MachineID == machineId {
```

Comment:
I like this idea to compare the machine-id.

But I wonder if that would impact the startup performance when `ig` is used outside of Kubernetes on a Linux machine, and I have a stale `$KUBECONFIG` that points to a API server that does not exist anymore. It would wait for the connection timeout.

You could argue that it is the user's fault to set `$KUBECONFIG` to a cluster that has been deleted... wdyt?

Reply:
I thought about it bit more and realized this can effect startup performance in case API Server is having issues which isn't great for `ig`. I am thinking of having following option in `containercollection`:

```
func WithCurrentNodeName() ContainerCollectionOption {
	return func(cc *ContainerCollection) error {
		go func() {
			nodeName, err := k8sutil.GetCurrentNode()
			if err != nil {
				log.Debugf("Failed to get current node name: %s", err)
				return
			}
			cc.mu.Lock()
			cc.nodeName = nodeName
			cc.mu.Unlock()
			return
		}()
		return nil
	}
}
```
Some initial events might miss node information but it is better to this way then to impact start. WDYT?

Reply:
Makes sense to handle it with #1992, that feels like a right way to do it. I can put the `os.Hostname()` (which isn't really `k8s.node`) for now and can deal with this after #1992